core/field_variable.js

```diff
@@ -47,15 +47,24 @@ goog.require('goog.string');
  * @constructor
  */
 Blockly.FieldVariable = function(varname, opt_validator, opt_variableTypes) {
-  Blockly.FieldVariable.superClass_.constructor.call(this,
-      Blockly.FieldVariable.dropdownCreate, opt_validator);
-  this.setValue(varname || '');
+  // The FieldDropdown constructor would call setValue, which might create a
+  // spurious variable.  Just do the relevant parts of the constructor.
+  this.menuGenerator_ = Blockly.FieldVariable.dropdownCreate;
+  this.size_ = new goog.math.Size(0, Blockly.BlockSvg.MIN_BLOCK_Y);
+  this.setValidator(opt_validator);
+  // TODO (#1499): Add opt_default_type to match default value.  If not set, ''.
+  this.defaultVariableName = (varname || '');
+  this.defaultType_ = '';
```

Comment:
I know there's a TODO here to add an opt_default_type, but I think for the time being, `defaultType_` can be smarter in the case where opt_variableTypes is actually provided and non-empty.

e.g. something like:

```
var hasVariableTypes = (opt__variableTypes && opt_variableTypes.length > 0);
this.defaultType_ = hasVariableTypes ? opt_variableTypes[0] : '';
```
OR

```
var hasSingleVariableType = (opt__variableTypes && opt_variableTypes.length == 1);
this.defaultType_ = hasSingleVariableType ? opt_variableTypes[0] : '';
```

Having the default type set correctly is definitely very useful when integrating with scratch-blocks.

Reply:
Oops, just realized we've already had this discussion before. You can ignore this.